### PR TITLE
Function case

### PR DESCRIPTION
**Pull request by pem70**
**What It Does**
Addresses [#315] by employing pylint for function name checks.

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: .github/workflows/sdk-build.yml
Comment (t1m0thyj) [inline .github/workflows/sdk-build.yml:37]: If we're adopting `pylint` then should we remove all references to `flakes8` rather than having a mix of both?

* Remove step that runs `flake8` in this workflow
* Remove `pyflakes` dependency from requirements.txt
* Update `.vscode/settings.json` to include setting that enables Pylint in VS Code 
* Replace `flakes8` with `pylint` in "Code Standards" section of CONTRIBUTING.md

File: CHANGELOG.md
Comment (traeok) [inline CHANGELOG.md:13]: Thoughts on making the changelog entry a bit more explicit for developers?

```suggestion
- *Breaking*: Revised function names in `Logger` class and `files` class into snake_case. Enabled pylint rule to enforce function case. [#315] (https://github.com/zowe/zowe-client-python-sdk/issues/315)
```

**Review by traeok — APPROVED** [review]
Changes LGTM, thanks @pem70!

File: .github/workflows/sdk-build.yml
Comment (traeok) [reply .github/workflows/sdk-build.yml:37]: I definitely think this is worth considering, but in the interest of avoiding scope creep, maybe this can be created as a separate issue?

Comment (zFernand0) [reply .github/workflows/sdk-build.yml:37]: https://github.com/zowe/zowe-client-python-sdk/issues/319

I'll leave it up to @pem70 to handle that issue as part of this PR or as part of a new one 😋

Comment (pem70) [reply .github/workflows/sdk-build.yml:37]: I will handle this as part of this PR

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @pem70! I ran system tests and they all passed 😋

**Review by zFernand0 — APPROVED** [review]
We should probably publish a new version after this PR is merged since it contains quite a few breaking changes. 😋 

That way, people have a chance to try things out before we publish a 1.0.0 (hopefully next week) 🙏🏽

**Pull request merged**
